ams/parser.js

```diff
@@ -65,6 +65,60 @@ module.exports = new Parser(function analyseEC(parsedUrl, ec) {
     if (param.doi) {
       result.unitid = param.doi.split('/')[1];
     }
+  } else if ((match = /^\/view\/journals\/([a-z]+)\/([0-9]+)\/([0-9]+)\/([a-z0-9-.]+)\.xml$/i.exec(path)) !== null && param.tab_body == 'pdf') {
+    // https://journals.ametsoc.org/view/journals/amsm/59/1/amsmonographs-d-18-0005.1.xml?tab_body=pdf
+    result.rtype = 'ARTICLE';
+    result.mime  = 'PDF';
+    result.title_id = match[1];
+    result.vol = match[2];
+    result.issue = match[3];
+    result.unitid = match[4];
+  } else if ((match = /^\/downloadpdf\/journals\/([a-z]+)\/([0-9]+)\/([0-9]+)\/([a-z0-9-.]+)\.xml$/i.exec(path)) !== null) {
+    // https://journals.ametsoc.org/view/journals/amsm/59/1/amsmonographs-d-18-0005.1.xml?tab_body=pdf
+    // https://journals.ametsoc.org/downloadpdf/journals/amsm/59/1/amsmonographs-d-18-0005.1.xml
+    result.rtype = 'ARTICLE';
+    result.mime  = 'PDF';
+    result.title_id = match[1];
+    result.vol = match[2];
+    result.issue = match[3];
+    result.unitid = match[4];
+
+  } else if ((match = /^\/view\/journals\/([a-z]+)\/([0-9]+)\/([0-9]+)\/([a-z0-9-.]+)\.xml$/i.exec(path)) !== null && (param.tab_body == 'fulltext-display' || param.tab_body == null)) {
+    // https://journals.ametsoc.org/view/journals/amsm/59/1/amsmonographs-d-18-0005.1.xml?tab_body=fulltext-display
+    // https://journals.ametsoc.org/view/journals/amsm/59/1/amsmonographs-d-18-0009.1.xml
+    result.rtype = 'ARTICLE';
+    result.mime  = 'HTML';
+    result.title_id = match[1];
+    result.vol = match[2];
+    result.issue = match[3];
+    result.unitid = match[4];
+
+  } else if ((match = /^\/view\/journals\/([a-z]+)\/([0-9]+)\/([0-9]+)\/[a-z]+-([a-z0-9-.]+)\.jpg$/i.exec(path)) !== null) {
+    // https://journals.ametsoc.org/view/journals/amsm/59/1/full-amsmonographs-d-18-0005.1-f2.jpg
+    result.rtype = 'IMAGE';
+    result.mime  = 'JPEG';
+    result.title_id = match[1];
+    result.vol = match[2];
+    result.issue = match[3];
+    result.unitid = match[4];
+
+  } else if ((match = /^\/view\/journals\/([a-z]+)\/([0-9]+)\/([0-9]+)\/([a-z0-9-.]+)\.xml$/i.exec(path)) !== null && param.tab_body == 'abstract-display') {
+    // https://journals.ametsoc.org/view/journals/amsm/59/1/amsmonographs-d-18-0005.1.xml?tab_body=abstract-display
+    result.rtype = 'ABS';
+    result.mime  = 'HTML';
+    result.title_id = match[1];
+    result.vol = match[2];
+    result.issue = match[3];
+    result.unitid = match[4];
+
+  } else if ((match = /^\/search$/i.exec(path)) !== null) {
+    // https://journals.ametsoc.org/search?q1=clouds
+    // https://journals.ametsoc.org/search?access=all&pageSize=10&q1=clouds&sort=relevance&t=apme
+    result.rtype = 'SEARCH';
+    result.mime  = 'HTML';
+    if (param.t) {
+      result.title_id = param.t;
+    }
```

Comment:
A ``SEARCH`` should only return the ``MIME`` and ``RTYPE``.

I come back to my suggestion of custom fields : [https://github.com/ezpaarse-project/ezpaarse-platforms/pull/477#discussion_r733696436](https://github.com/ezpaarse-project/ezpaarse-platforms/pull/477#discussion_r733696436)

After investigating the link, it appears that ``t`` can become ``t_0`` and that this always corresponds to a SEARCH.
Moreover, it is possible that ``t`` is empty and therefore the ``title_id`` would also be empty (the condition is not good enough).

Reply:
title_id is where you have had us put the "what set of content is this". We want to capture this on searches. So when a particular title is being searches we know this and can also correlate this with direct access of materials also associated with this title

Reply:
We can change the code to capture when the title is present in t_0 as well as t
```suggestion
    if (param.t) {
      result.title_id = param.t || param.t_0;
    }
```